lesson01.calculator/index.html

```diff
@@ -1,22 +1,318 @@
+      <button type="button" class="six">6</button>
+      <button type="button" class="seven">7</button>
+      <button type="button" class="eight">8</button>
+      <button type="button" class="nine">9</button>
+      <button type="button" class="zero">0</button>
+      <button type="button" class="point">.</button>
+      <button type="button" class="sign">+/-</button>
+      <button type="button" class="clear">C</button>
+      <button type="button" class="del">DEL</button>
+      <button type="button" class="divide">/</button>
+      <button type="button" class="multiply">x</button>
+      <button type="button" class="minus">-</button>
+      <button type="button" class="plus">+</button>
+      <button type="button" class="result">=</button>
+    </div>
+  </main>
+  <script>
+    'use strict';
+
+    const calc = (function() {
+
+      let Operator = {
+        add: 1,
+        subtract: 2,
+        multiply: 3,
+        divide: 4,
+      };
+
+      let _accumulator = 0;
+      let _operator;
+
+      //#region Public API Implementation
+      function doCalc(operand, operator) {
+        switch (_operator) {
+          case Operator.add:
+            _accumulator = _accumulator + Number(operand);
+            break;
+          case Operator.subtract:
+            _accumulator = _accumulator - Number(operand);
+            break;
+          case Operator.multiply:
+            _accumulator = _accumulator * Number(operand);
+            break;
+          case Operator.divide:
+            _accumulator = _accumulator / Number(operand);
+            break;
+          default:
+            _accumulator = Number(operand);
+            break;
+        }
+
+        setOperator(operator);
+
+        return _accumulator;
+      }
+
+      function clear() {
+        _accumulator = 0;
+        _operator = undefined;
+      }
+      //#endregion Public API Implementation
+
+      //#region Helper Functions
+      function setOperator(operator) {
+        switch (operator) {
+          case '+':
+            _operator = Operator.add;
+            break;
+          case '-':
+            _operator = Operator.subtract;
+            break;
+          case 'x':
+            _operator = Operator.multiply;
+            break;
+          case '/':
+            _operator = Operator.divide;
+            break;
+          case '=':
+            _operator = undefined;
+            break;
+          default:
+            break;
+        }
+      }
+      //#endregion Helper Functions
+
+      //#region Public API Declaration
+      let publicAPI = {
+        doCalc: doCalc,
+        clear: clear,
+      };
+      //#endregion Public API Declaration
+
+      return publicAPI;
+
+    })();
+
+    const MAX_NUMBER_LENGTH = 15;
+    const DEFAULT_DISPLAY_FONT_SIZE = '10'; // 10vmin
+
+    let displayElement;
+    let isCalcDone = false;
+
+    document.addEventListener("DOMContentLoaded", function(event) {
+      displayElement = document.querySelector("#display");
```

Comment:
A better way would be to give numbers, operators and other buttons a class, and assign a separate function for each group using `querySelectorAll('.someClass')`